api/chaos/serverexperimentation/v1/serverexperimentation.proto

```diff
@@ -148,6 +171,33 @@ message DownstreamEnforcing {
   }
 }
 
+// Enforce faults on upstream redis cluster.
+message RedisUpstreamEnforcing {
+  // An upstream responsible for enforcing a fault. Only requests that are handled
+  // by the specified upstream are considered for the fault injection.
+  oneof upstream_type {
+    option (validate.required) = true;
+
+    // A single upstream cluster.
+    SingleCluster upstream_cluster = 1;
+  }
+
+  // A downstream responsible for enforcing a fault. Only requests that are started
+  // by the specified downstream are considered for a fault injection.
+  oneof downstream_type {
+    option (validate.required) = true;
+
+    // A single downstream cluster.
+    SingleCluster downstream_cluster = 2;
+  }
+
+  // The redis commands to enforce a fault on. Passing in no redis commands, will apply faults on all redis commands.
```

Comment:
I'd simplify the language to "If unset, faults will apply to all redis commands."